### DISCUSSION TITLE
Has anyone added touch control to Quire (like pinch and zoom)?

### DISCUSSION THREAD

**Discussion by lizneely** [thegetty/quire#191] [Get Help] [answered]
Hi everyone!

I love how readable quire is on my phone and other touch screens (iPad, Surface), but it's not user-intuitive to use the control buttons for the zoomable images as we are now seemingly wired to pinch and zoom for a closer look. Has anyone added these controls to a quire template?

Any thoughts on this?
Thanks,
Liz

Liz Neely
Curator of Digital Experience
Georgia O'Keeffe Museum
Santa Fe, N.M.

**Comment by geealbers** [comment, marked as answer]
Hi @lizneely, we were getting quite a few user issues from people on phones trying to scroll down a page past a zooming image on the entry pages, and instead of scrolling the page they were inadvertently interacting with the image. We tried a few different fixes but then eventually intentionally disabled pinch zooming. Though you're right, it's not intuitive now. And I see on my latest test that the images still respond in other ways to touch, so we could have implemented that better.

We'll likely be replacing our zooming image viewer in the coming months so will look at this again then. In the meantime, I think this is the commit that disabled those touch interactions if you want to try reverse engineering it for your current project: https://github.com/thegetty/quire/commit/6aaf94f5e2d4ed9b1248a0534ae278460cd738f0.



https://github.com/thegetty/quire/commit/6aaf94f5e2d4ed9b1248a0534ae278460cd738f0

**Reply by daniel-keller** [reply]
Google Map's solution to the "scrolling past on mobile" problem is [cooperative gesture handling](https://developers.google.com/maps/documentation/javascript/interaction) which seems to work well. Leaflet doesn't include this by default but it looks like there is a [community built plugin](https://github.com/elmarquis/Leaflet.GestureHandling) to do the trick.
@geealbers if quire is moving away from leaflet do you know what tool you'll be using instead? I've been using openseadragon for our custom deepzoom but I'm considering using leaflet. With Openseadragon I'm struggling with the same design flaw.
